flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/hints/batch/NestLoopJoinHintTest.xml

```diff
@@ -679,8 +680,8 @@ HashJoin(joinType=[LeftSemiJoin], where=[=(a1, a2)], select=[a1, b1], build=[rig
    +- Calc(select=[a2])
       +- NestedLoopJoin(joinType=[InnerJoin], where=[=(a2, a3)], select=[a2, a3], build=[left])
          :- Exchange(distribution=[broadcast])
-         :  +- TableSourceScan(table=[[default_catalog, default_database, T2, project=[a2], metadata=[]]], fields=[a2], hints=[[[ALIAS options:[T2]]]])
-         +- TableSourceScan(table=[[default_catalog, default_database, T3, project=[a3], metadata=[]]], fields=[a3], hints=[[[ALIAS options:[T3]]]])
```

Comment:
Can we add test[s] to `ClearQueryBlockAliasResolverTest` with other hints than query block alias, to make sure that these hints are not removed

Reply:
Sure! I'll update it later

Reply:
Hi, I found ClearQueryBlockAliasResolverTest#testJoinHintWithJoinHintInCorrelateAndWithAgg do the similar test. Can you take a look to check if it's what you mean?
In this test, Join Hint will not be cleared and only Alias Hint will be cleared.
```
  <TestCase name="testJoinHintWithJoinHintInCorrelateAndWithAgg">
    <Resource name="sql">
      <![CDATA[select * from T1 WHERE a1 IN (select /*+ BrOaDcAsT(T2) */ count(T2.a2) from T2 join T1 on T2.a2 = T1.a1 group by T1.a1)]]>
    </Resource>
    <Resource name="ast">
      <![CDATA[
LogicalProject(a1=[$0], b1=[$1]), rowType=[RecordType(BIGINT a1, VARCHAR(2147483647) b1)]
+- LogicalFilter(condition=[IN($0, {
LogicalProject(EXPR$0=[$1]), rowType=[RecordType(BIGINT EXPR$0)]
+- LogicalAggregate(group=[{0}], EXPR$0=[COUNT($1)]), rowType=[RecordType(BIGINT a1, BIGINT EXPR$0)]
   +- LogicalProject(a1=[$2], a2=[$0]), rowType=[RecordType(BIGINT a1, BIGINT a2)]
      +- LogicalJoin(condition=[=($0, $2)], joinType=[inner], joinHints=[[[BROADCAST options:[LEFT]]]]), rowType=[RecordType(BIGINT a2, VARCHAR(2147483647) b2, BIGINT a1, VARCHAR(2147483647) b1)]
         :- LogicalTableScan(table=[[default_catalog, default_database, T2]]), rowType=[RecordType(BIGINT a2, VARCHAR(2147483647) b2)]
         +- LogicalTableScan(table=[[default_catalog, default_database, T1]]), rowType=[RecordType(BIGINT a1, VARCHAR(2147483647) b1)]
})]), rowType=[RecordType(BIGINT a1, VARCHAR(2147483647) b1)]
   +- LogicalTableScan(table=[[default_catalog, default_database, T1]]), rowType=[RecordType(BIGINT a1, VARCHAR(2147483647) b1)]
]]>
    </Resource>
  </TestCase>
```

Reply:
Hi, I mainly considered the other `hint`s that are part of `TableScan`. For example, 

```
LogicalSink(table=[default_catalog.default_database.ManagedTable], fields=[a, b, c], hints=[[[OPTIONS options:{compact.file-entries={"compact-partitions":[{"resolved-partition-spec":{"b":"0","c":"flink"},"file-entries":["/foo/bar/file"]}]}, ENRICHED_KEY=ENRICHED_VALUE, compact.file-base-path=/foo/bar}]]])
+- LogicalTableScan(table=[[default_catalog, default_database, ManagedTable]], hints=[[[OPTIONS inheritPath:[] options:{compact.file-entries={"compact-partitions":[{"resolved-partition-spec":{"b":"0","c":"flink"},"file-entries":["/foo/bar/file"]}]}, ENRICHED_KEY=ENRICHED_VALUE, compact.file-base-path=/foo/bar}]]])

```
. WDYT?

Reply:
Got it! I have added two tests for table hints.
1. There are some alias hints because of query hints exist, and table hints will not be cleared.
2. There are no some alias hints because of query hints not exist, and table hints will also not be cleared.